js/ai/src/prompt.ts

```diff
@@ -137,79 +174,381 @@ export interface ExecutablePrompt<
   asTool(): Promise<ToolAction>;
 }
 
+export type PartsGenerator<I, S = any> = (
+  input: I,
+  options: {
+    state?: S;
+  }
+) => Part[] | Promise<string | Part | Part[]>;
+
+export type MessageGenerator<I, S = any> = (
```

Comment:
I think `MessagesGenerator` to be consistent in plural naming. Though I might prefer `Resolver` to `Generator` for all of these names...

Reply:
Done.